### ISSUE TITLE
[MNG-6300] Multi module release creates empty directories in war file instead of jars

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7573]
**[Andreas Kurth](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=akurth)** opened **[MNG-6300](https://issues.apache.org/jira/browse/MNG-6300?redirect=false)** and commented

After updating to maven 3.5.2 we encounter the following reproducible bug with multi module builds.

If one of the modules is a war module and depends on another module, the dependency module will not be included as a jar file in WEB-INF/lib of the war file, but as an empty directory instead. Non module dependencies will be included correctly.

This bug does occur when the following conditions are met:

- running release:prepare/release:perform
- \<distributionManagement>\<site> element is present, so that release goals are "deploy site-deploy"
- \<reporting> element contains javadoc-maven-plugin

Please note that when running "mvn install" or "mvn deploy" the resulting war file is ok, while "mvn release:perform" creates corrupt files as described. Also, if javadoc-maven-plugin is not present in \<reporting> block, the war file is fine, too.

I have no idea whether this bug is maven core or rather release-plugin or even javadoc-plugin related, so I file it here. I prepared a minimal self contained example and attach it as mm.zip. To run the example, the following steps are needed:

```
cd /tmp
unzip /path/to/mm.zip
cd mm
git init
git add pom.xml mm-lib mm-war .gitignore
git commit
mvn release:prepare
mvn release:perform
```

After building the resulting corrupt war file can be found here:
repo/com/example/mm/mm-war/1.0.0/mm-war-1.0.0.war

---

**Affects:** 3.5.2

**Attachments:**
- [build.log](https://issues.apache.org/jira/secure/attachment/12896043/build.log) (_878.30 kB_)
- [mm.zip](https://issues.apache.org/jira/secure/attachment/12901035/mm.zip) (_6.52 kB_)

**Issue Links:**
- [MNG-8096](https://issues.apache.org/jira/browse/MNG-8096) Inconsistent dependency resolution behaviour for concurrent multi-module build can cause failures

- [MNG-6025](https://issues.apache.org/jira/browse/MNG-6025) Add a ProjectArtifactsCache similar to PluginArtifactsCache

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

It was kind of tricky to discover the issue, because it looks like any plugin interaction might already change the result.

But this seems to be the root cause:

```
be223808939d75152d7157db33d45f230114555e is the first bad commit
commit be223808939d75152d7157db33d45f230114555e
Author: Anton Tanasenko <atg.sleepless@gmail.com>
Date:   Wed May 18 18:40:28 2016 +0300

Add a ProjectArtifactsCache similar to PluginArtifactsCache

Signed-off-by: Jason van Zyl <jason@vanzyl.ca>
```

Must be fixed with the next release.

**Comment by jira-importer** [comment]
**[Andreas Kurth](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=akurth)** commented

Thanks Robert,

fix confirmed. I released one of our large company projects successfully.

Happily awaiting maven-3.5.3 now.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [98af937bc6298e82e489b5f0cd3fb0a1c3d37109](http://git-wip-us.apache.org/repos/asf/maven/commit/98af937b)
